nemo/core/classes/modelPT.py

```diff
@@ -169,7 +169,9 @@ def register_artifact(self, config_path: str, src: str):
                 used_src = src
             if not os.path.exists(used_src):
                 raise FileNotFoundError(f"Could not find {used_src}")
-            self.artifacts.append((config_path, used_src))
+            # But disregarding whether you use "local" or "remote" artifact - always store the original path.
+            # This fixes issues raising when finetuning NLP models that create and register tokenizer vocabs.
+            self.artifacts.append((config_path, src))
```

Comment:
@okuchaiev @ericharper @titu1994 @blisc @ekmb 

This is important for TLT fine-tuning where the (save-restore-save) cycle in NLP model tokenizers happen, can go with the full explanation "in person".

All tests are passing, still I think it is important to note this change.

Reply:
@okuchaiev despite green light will wait for your approval with merging this one as it is in "the core of the core".

Reply:
This needs to be used source only. The tests pass because all of the files exist in the test environment (in the original paths that were used for registration) - the real test is when you run this section on another users environment which doesn't contain those files. Blocking until proper review by @okuchaiev

Reply:
This area is outside of my knowledge with nemo. I can't really comment on this change.